engines/pink/archive.cpp

```diff
@@ -0,0 +1,327 @@
+		return new ConditionGameVariable;
+	case kConditionInventoryItemOwner:
+		return new ConditionInventoryItemOwner;
+	case kConditionModuleVariable:
+		return new ConditionModuleVariable;
+	case kConditionNotInventoryItemOwner:
+		return new ConditionNotInventoryItemOwner;
+	case kConditionNotModuleVariable:
+		return new ConditionNotModuleVariable;
+	case kConditionNotPageVariable:
+		return new ConditionNotPageVariable;
+	case kConditionPageVariable:
+		return new ConditionPageVariable;
+	case kCursorActor:
+		return new CursorActor;
+	case kGamePage:
+		return new GamePage;
+	case kHandlerLeftClick:
+		return new HandlerLeftClick;
+	case kHandlerStartPage:
+		return new HandlerStartPage;
+	case kHandlerTimer:
+	case kHandlerTimerActions:
+		return new HandlerTimerActions; // hack for Peril, but behavior is correct
+	case kHandlerTimerSequences:
+		return new HandlerTimerSequences;
+	case kHandlerUseClick:
+		return new HandlerUseClick;
+	case kInventoryActor:
+		return new InventoryActor;
+	case kInventoryItem:
+		return new InventoryItem;
+	case kLeadActor:
+		return new LeadActor;
+	case kModuleProxy:
+		return new ModuleProxy;
+	case kPDAButtonActor:
+		return new PDAButtonActor;
+	case kParlSqPink:
+		return new ParlSqPink;
+	case kPubPink:
+		return new PubPink;
+	case kSeqTimer:
+		return new SeqTimer;
+	case kSequence:
+		return new Sequence;
+	case kSequenceAudio:
+		return new SequenceAudio;
+	case kSequenceItem:
+		return new SequenceItem;
+	case kSequenceItemDefaultAction:
+		return new SequenceItemDefaultAction;
+	case kSequenceItemLeader:
+		return new SequenceItemLeader;
+	case kSequenceItemLeaderAudio:
+		return new SequenceItemLeaderAudio;
+	case kSideEffectExit:
+		return new SideEffectExit;
+	case kSideEffectGameVariable:
+		return new SideEffectGameVariable;
+	case kSideEffectInventoryItemOwner:
+		return new SideEffectInventoryItemOwner;
+	case kSideEffectLocation:
+		return new SideEffectLocation;
+	case kSideEffectModuleVariable:
+		return new SideEffectModuleVariable;
+	case kSideEffectPageVariable:
+		return new SideEffectPageVariable;
+	case kSideEffectRandomPageVariable:
+		return new SideEffectRandomPageVariable;
+	case kSupportingActor:
+		return new SupportingActor;
+	case kWalkAction:
+		return new WalkAction;
+	case kWalkLocation:
+		return new WalkLocation;
+	default:
+		error("Unknown object id");
+		return nullptr;
+	}
+}
+
+Archive::Archive(Common::SeekableReadStream  *stream)
+	: _readStream(stream), _writeStream(nullptr) {
+	_objectMap.push_back(0);
+	_objectIdMap.push_back(kNullObject);
+}
+
+Archive::Archive(Common::WriteStream *stream)
+	: _writeStream(stream), _readStream(nullptr) {
+	_objectMap.push_back(0);
+	_objectIdMap.push_back(kNullObject);
+}
+
+void Archive::mapObject(Object *obj) {
+	_objectMap.push_back(obj);
+	_objectIdMap.push_back(0);
+}
+
+int Archive::readCount() {
+	int count = _readStream->readUint16LE();
+
+	if (count == 0xffff) {
+		assert(0);
+		count = _readStream->readUint32LE();
```

Comment:
It looks like reading in as Uint32 is never done if the condition is true?

Reply:
Yes. It was to look if there were big arrays.
Removed this function completely

Reply:
Well, this was according to the specs. The code is coming from the fullpipe engine.